x-pack/test/upgrade/services/maps_upgrade_services.ts

```diff
@@ -13,18 +13,23 @@ export function MapsHelper({ getPageObjects, getService }: FtrProviderContext) {
   const log = getService('log');
 
   return {
-    async toggleLayerVisibility(layerName: string) {
-      log.debug('Inside toggleLayerVisibility');
-      await PageObjects.maps.openLayerTocActionsPanel(layerName);
-      await testSubjects.click('layerVisibilityToggleButton');
-      await PageObjects.common.sleep(3000);
+    async clearLegendTooltip() {
       const isTooltipOpen = await testSubjects.exists(`layerTocTooltip`, { timeout: 5000 });
       if (isTooltipOpen) {
         await testSubjects.click(`layerTocTooltip`);
         await PageObjects.common.sleep(1000);
       }
     },
 
+    async toggleLayerVisibility(layerName: string) {
```

Comment:
how about using gis_page toggleLayerVisibility and adding clearLegendTooltip to gis_page?

Reply:
@nreese I was thinking if the other maps tests are not failing, it's best not to add these changes.  In one of our last reviews we decided to create a maps service for upgrade, so I just added it here instead.  Thoughts?

Reply:
I would prefer to avoid duplicated functionality between gis_page and maps_upgrade_services.

There is also value in moving clearLegendTooltip to gis_page because another test, https://github.com/elastic/kibana/blob/main/x-pack/test/functional/apps/maps/group1/layer_visibility.js#L35, has similar logic.

Reply:
@nreese I moved the method pending testing.